File: b_housing_price/schemas/property.py
from pydantic import BaseModel, Field, field_validator

class propertyInput(BaseModel):
    square_footage: float = Field(..., gt=0, description="Square footage")
    bedrooms: int = Field(..., ge=1, le=20)
    bathrooms: int = Field(..., ge=1, le=3)
    year_built: int = Field(..., ge=1800, le=2025)
    lot_size: float = Field(..., gt=0)
    distance_to_city_center: float = Field(..., ge=0)
    school_rating: float = Field(..., ge=0, le=10)

    @field_validator('square_footage')
    def validate_sqft(cls, v):
        if v > 100000:
            raise ValueError('Square footage seems too large')
        return v

class propertyOutput(BaseModel):
    predicted_price: float
    input_data: propertyInput
    confidence: str

File: types/property.ts
export interface PropertyInput {
  square_footage: number;
  bedrooms: number;
  bathrooms: number;
  year_built: number;
  lot_size: number;
  distance_to_city_center: number;
  school_rating: number;
}

export interface PropertyOutput {
  predicted_price: number;
  input_data: PropertyInput;
  confidence: string;
}

export interface HistoryItem {
  id: string;
  timestamp: string;
  input: PropertyInput;
  result: PropertyOutput;
}

export interface CompareItem {
  id: string;
  label: string;
  input: PropertyInput;
  result: PropertyOutput;
}